eng/testing/ILLink.Descriptor.xunit.xml

```diff
@@ -15,5 +15,10 @@
     <namespace fullname="Xunit" />
     <namespace fullname="Xunit.Sdk" />
   </assembly>
+  <assembly fullname="xunit.assert">
+    <type fullname="Xunit.Sdk.AssertEqualityComparer`1">
+      <method signature="System.Boolean CompareTypedSets(System.Collections.IEnumerable,System.Collections.IEnumerable)" />
```

Comment:
Why is this necessary? Is xunit using Reflection to call this?

Reply:
Sorry, I missed the original description, I see it is using reflection.

What do you think about also going and fixing xunit in this case? At a minimum we could add a `DynamicDependency` there.

Reply:
https://github.com/xunit/assert.xunit/blob/58a59628853ed7a63895945fdac7984a92da2343/Sdk/AssertEqualityComparer.cs#L311-L313

From what I've seen in the past, attempting to annotate xUnit for trimming would be a health risk to the person doing it.

Reply:
I would just add xunit.assert to the list above, together with Xunit and Xunit.Sdk to be sure.

Reply:
Do you mean to preserve the whole `xunit.assert` assembly, instead of the specific method?

Reply:
Thanks for the detailed explanation, @MichalStrehovsky. I was a little bit hesitate to preserve the whole `xunit.assert` assembly, because these tests would also be running with AOT on WASM. AOT things on WASM is quite slow currently. My current philosophy is to preserve as little as possible to make the test pass. That's being said, I am also curious if preserving the whole `xunit.assert` assembly could fix more trimming related test failures. I will open another PR to try it out. For this PR, I would like to keep it the minimal.